code/go/0chain.net/smartcontract/dbs/event/stakepool.go

```diff
@@ -138,6 +138,10 @@ func (edb *EventDb) rewardUpdate(spus []dbs.StakePoolReward, round int64) error
 		if err := edb.rewardProviders(rewards.rewards, rewards.totalRewards, round); err != nil {
 			return fmt.Errorf("could not rewards providers: %v", err)
 		}
+
+		if err := edb.blobberSpecificRevenue(spus); err != nil {
```

Comment:
move this code a level higher, we should not keep any provider specific code here, since stakepool.go contains only common code, in process.go you can call different handler for this blobber reward